src/InteractiveWindow/Editor/Output/SortedSpans.cs

```diff
@@ -50,19 +50,36 @@ public IEnumerable<Span> GetOverlap(Span span)
 
             lock (_mutex)
             {
-                int startIndex = _spans.BinarySearch(span, comparer);
-                if (startIndex < 0)
+                var count = _spans.Count;
+
+                // _span is empty, no overlap.
+                if (count == 0)
                 {
-                    startIndex = ~startIndex - 1;
+                    return Enumerable.Empty<Span>();
                 }
 
+                int startIndex = _spans.BinarySearch(span, comparer);
+
+                // If span is not found in _span, BinarySearch returns a negative number that is the 
+                // bitwise complement of the index of the next span with larger starting index, or if
+                // there is no such span, the bitwise complement of _span.Count.   
+                
+                // Try get the span before the one with next larger starting index on in the list,
+                // unless the first span is the next larger one, then we just get the first one. 
                 if (startIndex < 0)
                 {
-                    return Enumerable.Empty<Span>();
+                    startIndex = ~startIndex;
                 }
+                                                       
+                if (startIndex > 0)
+                {
+                    startIndex = startIndex - 1;
+                }
```

Comment:
Is this case needed always or only when ```BinarySearch returns a negative number```? In other words, should this move inside the ```if``` above?